File: src/main/scala/io/corbel/sdk/notifications/Notifications.scala
package io.corbel.sdk.notifications

import io.corbel.sdk.auth.AuthenticationProvider._
import io.corbel.sdk.error.ApiError

import scala.collection.mutable
import scala.concurrent.{ExecutionContext, Future}

/**
  * @author Alexander De Leon (alex.deleon@devialab.com)
  */
trait Notifications {

  def sendNotification(id: String, recipient: String, properties: mutable.Map[String, String])(implicit authenticationProvider: AuthenticationProvider = null, ec: ExecutionContext): Future[Either[ApiError, Unit]]

}

File: src/main/java/io/corbel/sdk/java/notifications/Notifications.java
package io.corbel.sdk.java.notifications;

import io.corbel.sdk.error.ApiError;
import scala.Unit;
import scala.util.Either;

import java.util.Map;
import java.util.concurrent.CompletionStage;

/**
 * @author Ismael Madirolas (ismael.madirolas@devialab.com)
 */
public interface Notifications {

    CompletionStage<Either<ApiError, Unit>> sendNotification(String id, String recipient, Map<String, String> properties);
}
